### PR TITLE
Fix CMake build warnings

### LINKED ISSUE #359: CMake build issues
When building with CMake on Linux I encounter the following issues:

1.
```
CMake Deprecation Warning at libmaxminddb/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.
```
See https://cmake.org/cmake/help/latest/release/3.31.html#deprecated-and-removed-features.

2.
```
IPC::System::Simple required for Fatalised/autodying system() at libmaxminddb/dev-bin/make-man-pages.pl line 5.
	main::BEGIN() called at libmaxminddb/dev-bin/make-man-pages.pl line 5
	eval {...} called at libmaxminddb/dev-bin/make-man-pages.pl line 5
BEGIN failed--compilation aborted at libmaxminddb/dev-bin/make-man-pages.pl line 5.
```

The build proceeds smoothly otherwise. I am using CMake version 3.31.2.

### PR DESCRIPTION
**Pull request by oschwald**
Closes #359.

### CODE REVIEW

File: dev-bin/make-man-pages.pl [hunk 1]
```diff
@@ -67,18 +66,20 @@ sub _make_man {
...
+        ) == 0 or die "Failed to run lowdown: $!";
     }
 }
 
 sub _make_lib_man_links {
     my $target = shift;
 
-    my $header = read_file("$Bin/../include/maxminddb.h");
+    my $header = read_file("$Bin/../include/maxminddb.h")
+        or die "Failed to read header file: $!";
```
Comment (horgh) [inline dev-bin/make-man-pages.pl:77]: It looks to me like `read_file` might already handle errors for us.

File: dev-bin/make-man-pages.pl [hunk 2]
```diff
@@ -92,8 +93,8 @@ sub _pandoc_postprocess {
             s/^\.IP\n\.nf/.IP "" 4\n.nf/gm;
             s/(Automatically generated by Pandoc)(.+)$/$1/m;
         },
-        $file
-    );
+        $file,
+    ) or die "Failed to edit file: $!";
```
Comment (horgh) [inline dev-bin/make-man-pages.pl:97]: Likewise for `edit_file` as with `read_file`.

File: dev-bin/make-man-pages.pl [hunk 3]
```diff
@@ -67,18 +66,20 @@ sub _make_man {
...
     my $target = shift;
 
-    my $header = read_file("$Bin/../include/maxminddb.h");
+    my $header = read_file("$Bin/../include/maxminddb.h")
+        or die "Failed to read header file: $!";
     for my $proto ( $header =~ /^ *extern.+?(MMDB_\w+)\(/gsm ) {
-        open my $fh, '>', "$target/man/man3/$proto.3";
+        open my $fh, '>', "$target/man/man3/$proto.3"
+            or die "Failed to open file: $!";
         print {$fh} ".so man3/libmaxminddb.3\n";
```
Comment (horgh) [inline dev-bin/make-man-pages.pl:81]: I think `print` could fail too, but it looks like `autodie` already didn't handle that!

File: dev-bin/make-man-pages.pl [hunk 4]
```diff
@@ -41,7 +40,7 @@ sub _make_man {
 
     my $input   = "$Bin/../doc/$name.md";
     my $man_dir = "$target/man/man$section";
-    mkpath($man_dir);
+    mkpath($man_dir) or die "Failed to create directory $man_dir: $!";
```
Comment (horgh) [inline dev-bin/make-man-pages.pl:43]: Possibly `mkpath` handles errors already as well, although its docs seem more ambiguous to me!

**Review by horgh — CHANGES_REQUESTED** [review]

File: dev-bin/make-man-pages.pl (hunk 1 above)
Comment (oschwald) [reply dev-bin/make-man-pages.pl:77]: Ah, yeah, at the very least autodie didn't cover them. Sorry about that.

**Review by horgh — APPROVED** [review]

**Comment by gmou3** [comment]
Thanks for the quick response! However, now another issue has been unraveled for me:
```
Can't locate File/Slurp.pm in @INC (you may need to install the File::Slurp module) (@INC entries checked: /home/giorgos/perl5/lib/perl5 /usr/lib/perl5/5.40/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/5.40/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/5.40/core_perl /usr/share/perl5/core_perl) at libmaxminddb/dev-bin/make-man-pages.pl line 9.
BEGIN failed--compilation aborted at libmaxminddb/dev-bin/make-man-pages.pl line 9.
```
By the way, should the man pages always be generated? It took some time for me (after installing the File::Slurp module that is). Could a flag to choose whether to generate them be useful?

**Comment by oschwald** [comment]
If you use the released source packages, e.g., [libmaxminddb-1.11.0.tar.gz](https://github.com/maxmind/libmaxminddb/releases/download/1.11.0/libmaxminddb-1.11.0.tar.gz), you will not need to generate the man pages. We don't check in generated files into the repo.

**Comment by gmou3** [comment]
I am using the repo as a submodule and I build it using CMake. I would prefer it if people who built my project wouldn't get warnings (now there still is one due to the nonstandard `File::Slurp` perl module).

**Comment by oschwald** [comment]
We don't have plans to commit generated files to the repo. I think we would accept a PR to disable man page generation.

I'll go ahead and make a small PR to remove the dependency on non-core Perl modules as that seems beneficial generally, but the man-page generation will still depend on `lowdown` or `pandoc` being installed.

**Comment by gmou3** [comment]
Thanks again, seems to be working as intended! Apparently (from code inspection), it checks and handles the absence of `lowdown` and `pandoc`, so all is fine now.

**Pull request merged**